diff --git a/plugin/GreekCity.cfg b/plugin/GreekCity.cfg
new file mode 100644
index 0000000..e0a788d
--- /dev/null
+++ b/plugin/GreekCity.cfg
@@ -0,0 +1,1207 @@
+Greece/Athina
+Greece/Salonica
+Greece/Patra
+Greece/Iraklion
+Greece/Larissa
+Greece/Volos
+Greece/Hania
+Greece/Kavala
+Greece/Ioannina
+Greece/Halkida
+Greece/Serres
+Greece/Trikala
+Greece/Katerini
+Greece/Pireas
+Greece/Agrinion
+Greece/Alexandreia
+Greece/Alexandroupolis
+Greece/Arta
+Greece/Chios
+Greece/Corfu
+Greece/Didimotiho
+Greece/Drama
+Greece/Edessa
+Greece/Giannitsa
+Greece/Grevena
+Greece/Ierapetra
+Greece/Igoumenitsa
+Greece/Ithaka
+Greece/Kalamata
+Greece/Kalampaka
+Greece/Karpenisi
+Greece/Karditsa
+Greece/Kastoria
+Greece/Kilkis
+Greece/Komotini
+Greece/Korinthos
+Greece/Kos
+Greece/Kozani
+Greece/Lamia
+Greece/Lefkada
+Greece/Megalopoli
+Greece/Mikonos
+Greece/Mitilini
+Greece/Naousa
+Greece/Nafplio
+Greece/Naxos
+Greece/Paros
+Greece/Preveza
+Greece/Poligiros
+Greece/Samos
+Greece/Skiathos
+Greece/Skopelos
+Greece/Sparti
+Greece/Thasos
+Greece/Thiva
+Greece/Rethimno
+Greece/Veroia
+Greece/Xanthi
+Greece/Zante
+Greece/Abelakia
+Greece/Abelies
+Greece/Aegina
+Greece/Aigio
+Greece/Ayia_Anna
+Greece/Agia_Varvara
+Greece/Ayia_Galini
+Greece/Agia_Kiriaki
+Greece/Agia_Marina
+Greece/Agia_Paraskevi
+Greece/Agia_Trias
+Greece/Agiasma
+Greece/Agalas
+Greece/Agkathia
+Greece/Agelohori
+Greece/Angelokastron
+Greece/Agia
+Greece/Agiasos
+Greece/Agiofilo
+Greece/Agnanteron
+Greece/Agnantia
+Greece/Agria
+Greece/Aghialos
+Greece/Adamas
+Greece/Ardanion
+Greece/Athikia
+Greece/Aiandio
+Greece/Aiani
+Greece/Aigaleo
+Greece/Aigani
+Greece/Aeginio
+Greece/Aitoliko
+Greece/Afantou
+Greece/Afration
+Greece/Akharnai
+Greece/Akrata
+Greece/Akraifnion
+Greece/Akrini
+Greece/Akrolimni
+Greece/Akropotamia
+Greece/Aliartos
+Greece/Alepou
+Greece/Aliverion
+Greece/Alistrati
+Greece/Almyros
+Greece/Alfiousa
+Greece/Alonia
+Greece/Amarion
+Greece/Amarinthos
+Greece/Amalias
+Greece/Amarousion
+Greece/Amfissa
+Greece/Amisiana
+Greece/Ammoudia
+Greece/Ammokhorion
+Greece/Ammotopos
+Greece/Ambelokipoi
+Greece/Ambelon
+Greece/Amfiklia
+Greece/Amfilohia
+Greece/Anavra
+Greece/Anavisos
+Greece/Analipsis
+Greece/Aneza
+Greece/Anilion
+Greece/Anayennisis
+Greece/Anarrakhi
+Greece/Anatoli
+Greece/Anatolikon
+Greece/Andritsaina
+Greece/Andravida
+Greece/Andromahi
+Greece/Anthili
+Greece/Anthousa
+Greece/Antimahia
+Greece/Anifion
+Greece/Anogia
+Greece/Apollonia
+Greece/Apollona
+Greece/Arahova
+Greece/Arethousa
+Greece/Arisvi
+Greece/Aravisos
+Greece/Arakhnaion
+Greece/Argalasti
+Greece/Argos_Orestikon
+Greece/Argostolion
+Greece/Argirades
+Greece/Aryiropoulion
+Greece/Areopolis
+Greece/Aridaia
+Greece/Arkalohori
+Greece/Armenion
+Greece/Armenokhorion
+Greece/Armolia
+Greece/Arnaia
+Greece/Arnissa
+Greece/Arriana
+Greece/Arsenion
+Greece/Artesianon
+Greece/Arfara
+Greece/Arhaggelos
+Greece/Arkhaia_Korinthos
+Greece/Archaia_Olympia
+Greece/Arhontohori
+Greece/Asimion
+Greece/Asini
+Greece/Asvestohori
+Greece/Askos
+Greece/Asprovalta
+Greece/Asprokklisia
+Greece/Aspropirgos
+Greece/Astakos
+Greece/Astipalea
+Greece/Asopia
+Greece/Asopos
+Greece/Atalanti
+Greece/Atsiki
+Greece/Atsipopoulon
+Greece/Axioupolis
+Greece/Axos
+Greece/Bafra
+Greece/Benitsai
+Greece/Bouka
+Greece/Dafni
+Greece/Daratsos
+Greece/Dasohori
+Greece/Dafnon
+Greece/Delvinakion
+Greece/Deskati
+Greece/Dhelfoi
+Greece/Dhervenion
+Greece/Dikea
+Greece/Dimitra
+Greece/Dipotamia
+Greece/Dion
+Greece/Dimitsana
+Greece/Diava
+Greece/Diavata
+Greece/Diavatos
+Greece/Dialehto
+Greece/Dilinata
+Greece/Dolihi
+Greece/Doliana
+Greece/Domokos
+Greece/Drakotripa
+Greece/Drosaton
+Greece/Drosohori
+Greece/Drimos
+Greece/Ebesos
+Greece/Efkarpia
+Greece/Eirinoupoli
+Greece/Ekali
+Greece/Ekhinos
+Greece/Elatia
+Greece/Elati
+Greece/Elaiokhorion
+Greece/Elasson
+Greece/Elatohori
+Greece/Eleousa
+Greece/Eleftherion
+Greece/Eleftheroupoli
+Greece/Elefsis
+Greece/Eleon
+Greece/Ellinikon
+Greece/Emmanouil_Pappas
+Greece/Emborion
+Greece/Epano_Arhanai
+Greece/Episkepsis
+Greece/Epanomi
+Greece/Epiniana
+Greece/Episkopi
+Greece/Epitalion
+Greece/Eptahori
+Greece/Eratira
+Greece/Eretria
+Greece/Eratinon
+Greece/Eresos
+Greece/Ermioni
+Greece/Ermoupolis
+Greece/Erithre
+Greece/Esovalta
+Greece/Evropos
+Greece/Evzonoi
+Greece/Examilia
+Greece/Exaplatanos
+Greece/Exohi
+Greece/Falana
+Greece/Fanai
+Greece/Faros
+Greece/Farsa
+Greece/Farsala
+Greece/Fiki
+Greece/Filiro
+Greece/Fanarion
+Greece/Faskomilia
+Greece/Ferai
+Greece/Filiatai
+Greece/Filiatra
+Greece/Filippias
+Greece/Fillira
+Greece/Filotion
+Greece/Filotas
+Greece/Filla
+Greece/Fili
+Greece/Fitiai
+Greece/Fotolivos
+Greece/Flambouron
+Greece/Floyita
+Greece/Fourna
+Greece/Gazion
+Greece/Gazoros
+Greece/Gefira
+Greece/Gavalou
+Greece/Galatista
+Greece/Galatsion
+Greece/Galatades
+Greece/Galatas
+Greece/Galatini
+Greece/Galia
+Greece/Gallikos
+Greece/Gargalianoi
+Greece/Gardikion
+Greece/Gastouni
+Greece/Gastourion
+Greece/Gelanthi
+Greece/Gerania
+Greece/Gianades
+Greece/Gliki
+Greece/Glossa
+Greece/Gorgopi
+Greece/Gouvia
+Greece/Goumenissa
+Greece/Gouria
+Greece/Goumeron
+Greece/Gournai
+Greece/Griva
+Greece/Gra_Liyia
+Greece/Graikochorion
+Greece/Grammatikou
+Greece/Grammatikon
+Greece/Grammenitsa
+Greece/Granitsa
+Greece/Gimnotopos
+Greece/Gomfi
+Greece/Gonnoi
+Greece/Githio
+Greece/Herso
+Greece/Halandri
+Greece/Halastra
+Greece/Halkiades
+Greece/Halkidona
+Greece/Hariessa
+Greece/Holargos
+Greece/Hortata
+Greece/Hortiatis
+Greece/Hrisi
+Greece/Hrysovitsa
+Greece/hrysomilea
+Greece/Hrisohiri
+Greece/Hrysoupoli
+Greece/Horisti
+Greece/horygi
+Greece/Hora
+Greece/hora_Sfakion
+Greece/Ialissos
+Greece/Ierissos
+Greece/Inoi
+Greece/Inofita
+Greece/Isthmia
+Greece/Istiaia
+Greece/Itea
+Greece/Kavos
+Greece/Kalamos
+Greece/Kalimnos
+Greece/Kantanos
+Greece/Karpathos
+Greece/Karistos
+Greece/Kastron
+Greece/Kato_Agios_Ioannis
+Greece/Kato_Assos
+Greece/Kato_Apostoli
+Greece/Kato_Asitai
+Greece/Kata_Ahaia
+Greece/Kato_Garouna
+Greece/Kato_Glikovrisi
+Greece/Kato_Kamila
+Greece/Kato_Korakiana
+Greece/Kato_Lekhonia
+Greece/Kato_Mazarakion
+Greece/Kato_Milea
+Greece/Kato_Mitrousion
+Greece/Kato_Raptopoulon
+Greece/Kato_Soulion
+Greece/Kato_Skholarion
+Greece/Kato_Tithorea
+Greece/Kato_Khorion
+Greece/Kedros
+Greece/Kella
+Greece/Kefalos
+Greece/Kirra
+Greece/Kissamos
+Greece/Kitros
+Greece/Kavasila
+Greece/Kavasilas
+Greece/Kavallarion
+Greece/Kavvadades
+Greece/Kavili
+Greece/Kenourgio
+Greece/Kaissariani
+Greece/Kala_Nera
+Greece/Kalavrita
+Greece/Kali
+Greece/Kali_Vrisi
+Greece/Kalamakion
+Greece/Kalamaria
+Greece/Kalamia
+Greece/Kalambaki
+Greece/Kalamoti
+Greece/Kalamoton
+Greece/Kallithiron
+Greece/Kallifitos
+Greece/Kallithea
+Greece/Kallimasia
+Greece/Kallifonion
+Greece/Kalloni
+Greece/Kalogerikon
+Greece/Kalogriana
+Greece/Kalohori
+Greece/Kalpakion
+Greece/Kalovatos
+Greece/Kalos_Agros
+Greece/Kalivai
+Greece/Kalivia_Poliyirou
+Greece/Kalivia
+Greece/Kalivia_Thorikou
+Greece/Kamarai
+Greece/Kamarion
+Greece/Kamena_Vourla
+Greece/Kaminia
+Greece/Kamarina
+Greece/Kamariotissa
+Greece/Kaminarata
+Greece/Kampanis
+Greece/Kampi
+Greece/Kampohori
+Greece/Kanalion
+Greece/Kanalia
+Greece/Kanallakion
+Greece/Kandila
+Greece/Kapandrition
+Greece/Kaparellion
+Greece/Karatoulas
+Greece/Karitsa
+Greece/Karaiskakis
+Greece/Karvounarion
+Greece/Kardamaina
+Greece/Kardamila
+Greece/Kardamas
+Greece/Kardia
+Greece/Karditsomagoula
+Greece/Karousades
+Greece/Karperi
+Greece/Karperon
+Greece/Karpokhorion
+Greece/Karterion
+Greece/Karia
+Greece/Karies
+Greece/Kariai
+Greece/Kariotissa
+Greece/Karoti
+Greece/Kassiopi
+Greece/Kastella
+Greece/Kastellion
+Greece/Kastaneai
+Greece/Kastaneri
+Greece/Kastania
+Greece/Kastaneai
+Greece/Kastanoussa
+Greece/Kastrakion
+Greece/Kastrion
+Greece/Kastritsa
+Greece/Katakolon
+Greece/Katastarion
+Greece/Katafigion
+Greece/Katahas
+Greece/Katouna
+Greece/Katsikas
+Greece/Kerion
+Greece/Keramoti
+Greece/Kerasohori
+Greece/Keratea
+Greece/Keratsinion
+Greece/Kerkini
+Greece/Kestrini
+Greece/Kefalohori
+Greece/Kefalovriso
+Greece/Khalki
+Greece/Khalandritsa
+Greece/Khalkoutsion
+Greece/Khaniotis
+Greece/Kharopon
+Greece/Khimarros
+Greece/Khersonisos
+Greece/Khrisokhorafa
+Greece/Khrisokastron
+Greece/Khrison
+Greece/Kipia
+Greece/Kipourio
+Greece/Kifisia
+Greece/Kiaton
+Greece/Kiverion
+Greece/Kimmeria
+Greece/Kineta
+Greece/Kithira
+Greece/Kimi
+Greece/Kimina
+Greece/Kiria
+Greece/Kirkizatai
+Greece/Klitos
+Greece/Kleidion
+Greece/Klimatia
+Greece/Koila
+Greece/Koimisis
+Greece/Koilas
+Greece/Kokkarion
+Greece/Kokkinion
+Greece/Kokkinohoma
+Greece/Kokkonion
+Greece/Kolindros
+Greece/Kolimvarion
+Greece/Kolkhikon
+Greece/Kommenon
+Greece/Komnina
+Greece/Kobitsi
+Greece/Kompotion
+Greece/Kontakeika
+Greece/Kontariotissa
+Greece/Kontokalion
+Greece/Kopanos
+Greece/Korisos
+Greece/Korinos
+Greece/Kormista
+Greece/Koromilea
+Greece/Korifi
+Greece/Koropion
+Greece/Koroni
+Greece/Kouvaras
+Greece/Kouklio
+Greece/Kouloura
+Greece/Koumeika
+Greece/Koutouloufarion
+Greece/Koutselion
+Greece/Koutsouras
+Greece/Koufalia
+Greece/Koufovounon
+Greece/Krentis
+Greece/Krestena
+Greece/Krini
+Greece/Kranea
+Greece/Kranea_Elassonos
+Greece/Kranidion
+Greece/Kranohori
+Greece/Kremasti
+Greece/Kristoni
+Greece/Krithia
+Greece/Kritsa
+Greece/Krokeai
+Greece/Krousonas
+Greece/Krioneri
+Greece/Kristallopigi
+Greece/Krokos
+Greece/Kria_Vrisi
+Greece/Killini
+Greece/Kinopiastes
+Greece/Kiparissia
+Greece/Kiprinos
+Greece/Kiras_Vrisi
+Greece/Kiriakion
+Greece/Konstantia
+Greece/Kostakioi
+Greece/Kostalexis
+Greece/Kokkinon
+Greece/Komara
+Greece/Konitsa
+Greece/Koustaros
+Greece/Lavara
+Greece/Lakkoma
+Greece/Lalas
+Greece/Lapas
+Greece/Lardos
+Greece/Larimna
+Greece/Leka
+Greece/Leheo
+Greece/Lithion
+Greece/Limnai
+Greece/Limni
+Greece/Lindos
+Greece/Lagada
+Greece/Lagadas
+Greece/Lagina
+Greece/Lagos
+Greece/Lakithra
+Greece/Lakkion
+Greece/Lakkopetra
+Greece/Lepenou
+Greece/Leptokaria
+Greece/Ieropigi
+Greece/Lefkadia
+Greece/Lefkimmi
+Greece/Lehena
+Greece/Leonidion
+Greece/Limerion
+Greece/Lixourion
+Greece/Liti
+Greece/Leianovergion
+Greece/Liapades
+Greece/Livadion
+Greece/Livadia
+Greece/Livadheron
+Greece/Livanata
+Greece/Lidorikion
+Greece/Lithakia
+Greece/Lithohori
+Greece/Limenas_Hersonisou
+Greece/Limenaria
+Greece/Linopotis
+Greece/Lipohori
+Greece/Litohoro
+Greece/Longa
+Greece/Logades
+Greece/Loudias
+Greece/Loukisia
+Greece/Loutra
+Greece/Loutra_Ipatis
+Greece/Loutrakion
+Greece/Loutron
+Greece/Loutros
+Greece/Louros
+Greece/Likoporia
+Greece/Likovrisi
+Greece/Lofos
+Greece/Likion
+Greece/Malia
+Greece/Mandalon
+Greece/Mandra
+Greece/Mandrai
+Greece/Marmara
+Greece/Marmaron
+Greece/Krionerion
+Greece/Mega_Kefalovriso
+Greece/Mega_Piston
+Greece/Megara
+Greece/Methana
+Greece/Metsovon
+Greece/Mithimna
+Greece/Milos
+Greece/Magoulitsa
+Greece/Magoula
+Greece/Mazarakia
+Greece/Makrisia
+Greece/Makrakomi
+Greece/Makrinitsa
+Greece/Makrohori
+Greece/Makrikapa
+Greece/Makrinitsa
+Greece/Makrikhorion
+Greece/Makrigialos
+Greece/Malesina
+Greece/Maniakoi
+Greece/Manolas
+Greece/Mantamados
+Greece/Mantoudion
+Greece/Marina
+Greece/Marathea
+Greece/Marathokampos
+Greece/Marathopolis
+Greece/Marathonas
+Greece/Margarition
+Greece/Markopoulon
+Greece/Markopoulon_Oropou
+Greece/Marmarion
+Greece/Martinon
+Greece/Mastihari
+Greece/Mavrovouni
+Greece/Mavrommation
+Greece/Mavronerion
+Greece/Mavroudion
+Greece/Megala_Kalivia
+Greece/Megali_Vrisi
+Greece/Megali_Panayia
+Greece/Megali_Khora
+Greece/Megalos_Prinos
+Greece/Megarhi
+Greece/Megalohori
+Greece/Megalokhorion
+Greece/Methoni
+Greece/Melivoia
+Greece/Meliki
+Greece/Melisi
+Greece/Melissia
+Greece/Meliti
+Greece/Meligalas
+Greece/Melisourgi
+Greece/Melissohori
+Greece/Menidion
+Greece/Menemeni
+Greece/Mesagros
+Greece/Mesaria
+Greece/Mesimeri
+Greece/Mesimerion
+Greece/Mesolongion
+Greece/Mesopotamia
+Greece/Mesopotamon
+Greece/Messini
+Greece/Mesta
+Greece/Metamorfosis
+Greece/Metaxades
+Greece/Metohi
+Greece/Milea
+Greece/Mileai
+Greece/Mitropolis
+Greece/Mikrolimni
+Greece/Mikromani
+Greece/Mikro_Derio
+Greece/Mikron_Monastirion
+Greece/Mikrokambos
+Greece/Mikropolis
+Greece/Milesion
+Greece/Mires
+Greece/Molaoi
+Greece/Monastirakion
+Greece/Monahiti
+Greece/Monospita
+Greece/Morfovounion
+Greece/Moshato
+Greece/Moshopotamos
+Greece/Moskhokhorion
+Greece/Mouzakion
+Greece/Mouzouras
+Greece/Mouria
+Greece/Mouriai
+Greece/Mourniai
+Greece/Moustheni
+Greece/Mokhos
+Greece/Miriofiton
+Greece/Mirsini
+Greece/Myrtea
+Greece/Mitilinioi
+Greece/Moria
+Greece/Morfion
+Greece/Miki
+Greece/Miloi
+Greece/Myrina
+Greece/Mitikas
+Greece/Molos
+Greece/Nea_Efesos
+Greece/Nea_Anhialos
+Greece/Nea_Alikarnassos
+Greece/Nea_Apollonia
+Greece/Nea_Artaki
+Greece/Nea_Vissa
+Greece/Nea_Erithraia
+Greece/Nea_Zihni
+Greece/Nea_Irakleia
+Greece/Nea_Iraklitsa
+Greece/Nea_Ionia
+Greece/Nea_Kios
+Greece/Nea_Kallikratia
+Greece/Nea_Karvali
+Greece/Nea_Karia
+Greece/Nea_Kerasia
+Greece/Nea_Kerasounta
+Greece/Nea_Makri
+Greece/Nea_Malgara
+Greece/Nea_Magnisia
+Greece/Nea_Manolas
+Greece/Nea_Mesimvria
+Greece/Nea_Mihaniona
+Greece/Nea_Nikomidia
+Greece/Nea_Pella
+Greece/Nea_Peramos
+Greece/Nea_Palatia
+Greece/Nea_Playia
+Greece/Nea_Santa
+Greece/Nea_Sampsus
+Greece/Nea_Selefkeia
+Greece/Nea_Sinopi
+Greece/Nea_Smirni
+Greece/Nea_Stira
+Greece/Nea_Tirins
+Greece/Nea_Triglia
+Greece/Nea_Filadelfia
+Greece/Nea_Fokaia
+Greece/Nenita
+Greece/Neon_Agionerion
+Greece/Neo_Argirio
+Greece/Neon_Ginekokastro
+Greece/Neo_Karlovasi
+Greece/Neo_Risi
+Greece/Neon_Soulion
+Greece/Neoi_Epivatai
+Greece/Neoi_Pori
+Greece/Neon_Monastirion
+Greece/Neon_Petritsion
+Greece/Neos_Marmaras
+Greece/Neos_Milotopos
+Greece/Neos_Pirgos
+Greece/Neos_Skopos
+Greece/Neos_Oropos
+Greece/Nikea
+Greece/Neapolis
+Greece/Nemea
+Greece/Neokaisaria
+Greece/Neohorouda
+Greece/Neohoropoulo
+Greece/Neohori
+Greece/Nerokouros
+Greece/Nestorion
+Greece/Neokastron
+Greece/Nisi
+Greece/Nigrita
+Greece/Nikisiani
+Greece/Nikoklia
+Greece/Niseli
+Greece/Nimfes
+Greece/Oia
+Greece/Oinoussai
+Greece/Oihalia
+Greece/Omvriaki
+Greece/Omolion
+Greece/Ovria
+Greece/Othoni
+Greece/Orestias
+Greece/Ormenion
+Greece/Ormilia
+Greece/Orhomenos
+Greece/Oraiokastron
+Greece/Oreoi
+Greece/Oropos
+Greece/Ouranopolis
+Greece/Oxilithos
+Greece/Oxinia
+Greece/Palairos
+Greece/Pamfylla
+Greece/Panormos
+Greece/Parga
+Greece/Pasion
+Greece/Pakhni
+Greece/Pagouria
+Greece/Pagondas
+Greece/Paiania
+Greece/Paleokastro
+Greece/Palefyto
+Greece/Palaia_Fokea
+Greece/Palaiokklision
+Greece/Palaiokomi
+Greece/Palaiomanina
+Greece/Paleomonastiro
+Greece/Paleostani
+Greece/Paleohora
+Greece/Palaiohori
+Greece/Palaiokhorion
+Greece/Paleo_Agioneri
+Greece/Palaio_Eleftherohori
+Greece/Palaion_Keramidion
+Greece/Palaion_Faliron
+Greece/Paleokastro
+Greece/Paleokipos
+Greece/Paleopirgos
+Greece/Palaios_Mylotopos
+Greece/Palamas
+Greece/Palatitsia
+Greece/Paliampela
+Greece/Paliouria
+Greece/Pallini
+Greece/Panagia
+Greece/Panetolio
+Greece/Pantokrator
+Greece/Panorama
+Greece/Oikismos_Papagou
+Greece/Pappados
+Greece/Pappayiannis
+Greece/Pappadatai
+Greece/Parakoila
+Greece/Paralion_Astros
+Greece/Paravola
+Greece/Parakalamos
+Greece/Paralia
+Greece/Paralimni
+Greece/Paramithia
+Greece/Parapotamos
+Greece/Patitirion
+Greece/Patris
+Greece/Pedini
+Greece/Pedinon
+Greece/Pella
+Greece/Pelasyia
+Greece/Pelopion
+Greece/Pentalofon
+Greece/Pentalofos
+Greece/Pendapolis
+Greece/Pentaplatanon
+Greece/Peplos
+Greece/Perama
+Greece/Perdika
+Greece/Perni
+Greece/Petas
+Greece/Petra
+Greece/Peristasis
+Greece/Perea
+Greece/Perakhora
+Greece/Perdikakion
+Greece/Perivolion
+Greece/Perivolia
+Greece/Perigialion
+Greece/Periyialion
+Greece/Peristerion
+Greece/Peristera
+Greece/Peroulades
+Greece/Petinos
+Greece/Petrea
+Greece/Petriti
+Greece/Petrokhorion
+Greece/Petroupolis
+Greece/Petroussa
+Greece/Petroton
+Greece/Petroporos
+Greece/Pefkodasos
+Greece/Pefka
+Greece/Pigai
+Greece/Pigi
+Greece/Pialeia
+Greece/Pitharion
+Greece/Pikermion
+Greece/Pithagorio
+Greece/Pilio
+Greece/Pilea
+Greece/Pirgi
+Greece/Pirgetos
+Greece/Pithio
+Greece/Pili
+Greece/Pilos
+Greece/Pirgos
+Greece/Plagia
+Greece/Planos
+Greece/Platanos
+Greece/Plagia
+Greece/Plagiarion
+Greece/Plataiai
+Greece/Platanakia
+Greece/Platanorrevma
+Greece/Plataria
+Greece/Platy
+Greece/Platikambos
+Greece/Plomarion
+Greece/Polihni
+Greece/Politika
+Greece/Polihnitos
+Greece/Polykarpion
+Greece/Poliplatanos
+Greece/Polidroso
+Greece/Polikastro
+Greece/Polimilos
+Greece/Polikhronon
+Greece/Pontoiraklia
+Greece/Portaria
+Greece/Portoheli
+Greece/Potamia
+Greece/Potamos
+Greece/Pournarion
+Greece/Pramanta
+Greece/Prinos
+Greece/Proastion
+Greece/Provatas
+Greece/Provaton
+Greece/Prokopion
+Greece/Prosotsani
+Greece/Profitis
+Greece/Profitis_Ilias
+Greece/Prodromos
+Greece/Promahoi
+Greece/Prohoma
+Greece/Proti
+Greece/Ptelea
+Greece/Pteleos
+Greece/Pteri
+Greece/Pteria
+Greece/Ptolemais
+Greece/Pogoniani
+Greece/Pombia
+Greece/Poros
+Greece/Psarades
+Greece/Psahna
+Greece/Psihiko
+Greece/Rahi
+Greece/Rizia
+Greece/Rizoma
+Greece/Rion
+Greece/Raptopoulon
+Greece/Rafina
+Greece/Rahona
+Greece/Rapsani
+Greece/Rentina
+Greece/Ritini
+Greece/Rizarion
+Greece/Rizareion
+Greece/Rizovounion
+Greece/Rizon
+Greece/Rizomilos
+Greece/Rizomata
+Greece/Roviai
+Greece/Rovoliari
+Greece/Royitika
+Greece/Rodotopion
+Greece/Rodopolis
+Greece/Rousson
+Greece/Riakia
+Greece/Rokka
+Greece/Rizia
+Greece/Sami
+Greece/Sapai
+Greece/Sarti
+Greece/Seleron
+Greece/Servia
+Greece/Sindos
+Greece/Sision
+Greece/Savalia
+Greece/Sagiada
+Greece/Salamis
+Greece/Samarina
+Greece/Samothraki
+Greece/Sarantaporon
+Greece/Sardinia
+Greece/Svoronos
+Greece/Sevasti
+Greece/Sevastiana
+Greece/Selinia
+Greece/Selianitika
+Greece/Selades
+Greece/Servota
+Greece/Sisamia
+Greece/Sitia
+Greece/Siatista
+Greece/Sidirokastron
+Greece/Sikion
+Greece/Sinarades
+Greece/Sitagroi
+Greece/Skala
+Greece/Skala_Oropou
+Greece/Skala_Fourkas
+Greece/Skalanion
+Greece/Skotina
+Greece/Skotoussa
+Greece/Skourta
+Greece/Skoutari
+Greece/Skriperon
+Greece/Skydra
+Greece/Skiros
+Greece/Smixi
+Greece/Souroti
+Greece/Souflion
+Greece/Sofikon
+Greece/Sohos
+Greece/Souda
+Greece/Sourpi
+Greece/Sparta
+Greece/Spata
+Greece/Spetsai
+Greece/Spilion
+Greece/Spatharaioi
+Greece/Spartilas
+Greece/Sperhiada
+Greece/Sperkhoyia
+Greece/Spolaita
+Greece/Stanos
+Greece/Sterna
+Greece/Sternai
+Greece/Stalis
+Greece/Stamata
+Greece/Stamna
+Greece/Stavraki
+Greece/Stavros
+Greece/Stirion
+Greece/Stenimahos
+Greece/Stefani
+Greece/Stefanavikion
+Greece/Stratos
+Greece/Stratonion
+Greece/Stilis
+Greece/Stomion
+Greece/Stipsi
+Greece/Sikea
+Greece/Sykeai
+Greece/Sikiada
+Greece/Sikouri
+Greece/Siliveniotika
+Greece/Sfendamion
+Greece/Skhisma
+Greece/Shimatari
+Greece/Sozopolis
+Greece/Sosandra
+Greece/Sivota
+Greece/Simi
+Greece/Taxiarhes
+Greece/Taxiarkhis
+Greece/Terpni
+Greece/Terpsithea
+Greece/Temeni
+Greece/Terpilos
+Greece/Thermi
+Greece/Thermon
+Greece/Thira
+Greece/Theodorakio
+Greece/Theodoriana
+Greece/Theologos
+Greece/Therianon
+Greece/Thespiai
+Greece/Thesprotikon
+Greece/Theopetra
+Greece/Tholopotamion
+Greece/Tholos
+Greece/Thouria
+Greece/Thrapsanon
+Greece/Thimiana
+Greece/Thirion
+Greece/Tinos
+Greece/Tihero
+Greece/Tilisos
+Greece/Tithorea
+Greece/Tolon
+Greece/Topoliana
+Greece/Toumba
+Greece/Trikomon
+Greece/Trilofon
+Greece/Trilofos
+Greece/Tripolis
+Greece/Tragana
+Greece/Traganon
+Greece/Triandria
+Greece/Trikeri
+Greece/Trikokkia
+Greece/Trifos
+Greece/Tsakoi
+Greece/Tsaritsani
+Greece/Tsikalaria
+Greece/Tsotilion
+Greece/Tympakion
+Greece/Tyrgia
+Greece/Tyrnavos
+Greece/Tzanata
+Greece/Tzermiadon
+Greece/Vayia
+Greece/Vamos
+Greece/Varda
+Greece/Vari
+Greece/Velo
+Greece/Veryi
+Greece/Vessa
+Greece/Vilia
+Greece/Vathi
+Greece/Vathilakkos
+Greece/Valanidorrahi
+Greece/Valsamata
+Greece/Valteron
+Greece/Valtinon
+Greece/Valtotopion
+Greece/Valira
+Greece/Vamvakou
+Greece/Vamvakofiton
+Greece/Vanaton
+Greece/Varvasaina
+Greece/Varia
+Greece/Vartholomion
+Greece/Varnavas
+Greece/Vasiliai
+Greece/Vasileonoiko
+Greece/Vasilika
+Greece/Vasiliki
+Greece/Vasiliko
+Greece/Vasilopoulo
+Greece/Vatsounia
+Greece/Vafiohori
+Greece/Velventos
+Greece/Velestinon
+Greece/Veneraton
+Greece/Vergina
+Greece/Verdikoussa
+Greece/Viros
+Greece/Vlahata
+Greece/Vlahiotis
+Greece/Vlakhoyiannion
+Greece/Vlakhokerasea
+Greece/Vlakhopoulon
+Greece/Volimai
+Greece/Volissos
+Greece/Voskohori
+Greece/Vouvopotamos
+Greece/Vourgareli
+Greece/Vouliagmeni
+Greece/Vounoplagia
+Greece/Vourliotai
+Greece/Voula
+Greece/Voulpi
+Greece/Vrina
+Greece/Vrahati
+Greece/Vrahneika
+Greece/Vrontados
+Greece/Vronteron
+Greece/Vrontou
+Greece/Vrisaki
+Greece/Vironia
+Greece/Vitina
+Greece/Vonitsa
+Greece/Viron
+Greece/Volax
+Greece/Xehasmeni
+Greece/Xirokambion
+Greece/Xiropotamos
+Greece/Xirohori
+Greece/Xirokambos
+Greece/Xinon_Neron
+Greece/Xifiani
+Greece/Xilagani
+Greece/Xylokastro
+Greece/Xilopolis
+Greece/Xinonerion
+Greece/Ydrousa
+Greece/Yeryeri
+Greece/Yenisea
+Greece/Yerakion
+Greece/Yeranion
+Greece/Yerakarou
+Greece/Yeoryiani
+Greece/Yiannouli
+Greece/Yimnon
+Greece/Zarakes
+Greece/Zarkos
+Greece/Zagora
+Greece/Zaros
+Greece/Zaharo
+Greece/Zelion
+Greece/Zitsa
+Greece/Zilefti
+Greece/Ziparion
+Greece/Zigos
+Greece/Zouzouli
+Greece/Zoniana
diff --git a/plugin/ui.py b/plugin/ui.py
index f3912ce..36953ae 100644
--- a/plugin/ui.py
+++ b/plugin/ui.py
@@ -13,7 +13,7 @@
 #
 from __future__ import absolute_import
 from __future__ import print_function
-VERSION = "3.2.6"
+VERSION = "3.2.6-gr.mod"
 #
 #                    04.10.2017
 #
@@ -183,7 +183,6 @@ config.plugins.foreca.units = ConfigSelection(default="metrickmh", choices=[("me
 config.plugins.foreca.time = ConfigSelection(default="24h", choices=[("12h", _("12 h")), ("24h", _("24 h"))])
 config.plugins.foreca.debug = ConfigEnableDisable(default=False)
 
-
 HEADERS = {'User-Agent': 'Mozilla/5.0 (SmartHub; SMART-TV; U; Linux/SmartTV; Maple2012) AppleWebKit/534.7 (KHTML, like Gecko) SmartTV Safari/534.7'}
 MAIN_PAGE = "http://www.foreca.net"
 USR_PATH = resolveFilename(SCOPE_CONFIG) + "Foreca"
@@ -436,6 +435,7 @@ class MainMenuList(MenuList):
 		else:
 			self.centigrades = int(self.x[2])
 			tempUnit = _(SIGN + "C")
+
 		if self.centigrades <= -20:
 			self.tempcolor = ddblau
 		elif self.centigrades <= -15:
@@ -469,7 +469,10 @@ class MainMenuList(MenuList):
 		pngpic = LoadPixmap(self.thumb)
 		if pngpic is not None:
 			x, y, w, h = self.valPict
-			self.res.append(MultiContentEntryPixmapAlphaTest(pos=(x, y), size=(w, h), png=pngpic))
+			flags = 0
+			if self.valPictScale:
+				flags = BT_SCALE | BT_KEEP_ASPECT_RATIO
+			self.res.append(MultiContentEntryPixmapAlphaTest(pos=(x, y), size=(w, h), png=pngpic, flags = flags ))
 
 		# Temp
 		x, y, w, h = self.valTemp
@@ -481,7 +484,10 @@ class MainMenuList(MenuList):
 		pngpic = LoadPixmap(self.wind + ".png")
 		if pngpic is not None:
 			x, y, w, h = self.valWindPict
-			self.res.append(MultiContentEntryPixmapAlphaTest(pos=(x, y), size=(w, h), png=pngpic))
+			flags = 0
+			if self.valWindPictScale:
+				flags = BT_SCALE | BT_KEEP_ASPECT_RATIO
+			self.res.append(MultiContentEntryPixmapAlphaTest(pos=(x, y), size=(w, h), png=pngpic, flags = flags))
 
 		# Wind
 		x, y, w, h = self.valWind
@@ -491,7 +497,7 @@ class MainMenuList(MenuList):
 
 		# Text
 		x, y, w, h = self.valText1
-		self.res.append(MultiContentEntryText(pos=(x, y), size=(w, h), font=3, text=_(self.x[5]), color=weiss, color_sel=weiss))
+		self.res.append(MultiContentEntryText(pos=(x, y), size=(w, h), font=3, text=self.x[5], color=weiss, color_sel=weiss))
 		x, y, w, h = self.valText2
 		textsechs = self.x[6]
 		textsechs = textsechs.replace("&deg;", "") + tempUnit
@@ -612,8 +618,8 @@ class ForecaPreview(Screen, HelpableScreen):
 
 		if HD:
 			self.skin = """
-				<screen name="ForecaPreview" position="center,center" size="980,505" title="Foreca Weather Forecast" backgroundColor="#00000000" >
-					<widget name="MainList" position="0,90" size="980,365" zPosition="3" backgroundColor="#00000000" enableWrapAround="1" scrollbarMode="showOnDemand" />
+				<screen name="ForecaPreview" position="center,center" size="980,505" title="Foreca Weather Forecast" backgroundColor="#40000000" >
+					<widget name="MainList" position="0,90" size="980,365" zPosition="3" backgroundColor="#40000000" enableWrapAround="1" scrollbarMode="showOnDemand" />
 					<widget source="Titel" render="Label" position="4,10" zPosition="3" size="978,60" font="Regular;24" valign="center" halign="left" transparent="1" foregroundColor="#ffffff"/>
 					<widget source="Titel2" render="Label" position="35,15" zPosition="2" size="900,60" font="Regular;26" valign="center" halign="center" transparent="1" foregroundColor="#f47d19"/>
 					<eLabel position="5,70" zPosition="2" size="970,2" foregroundColor="#c3c3c9" backgroundColor="#FFFFFF" />
@@ -636,8 +642,8 @@ class ForecaPreview(Screen, HelpableScreen):
 				</screen>"""
 		else:
 			self.skin = """
-				<screen name="ForecaPreview" position="center,65" size="720,480" title="Foreca Weather Forecast" backgroundColor="#00000000" >
-					<widget name="MainList" position="0,65" size="720,363" zPosition="3" backgroundColor="#00000000" enableWrapAround="1" scrollbarMode="showOnDemand" />
+				<screen name="ForecaPreview" position="center,65" size="720,480" title="Foreca Weather Forecast" backgroundColor="#40000000" >
+					<widget name="MainList" position="0,65" size="720,363" zPosition="3" backgroundColor="#40000000" enableWrapAround="1" scrollbarMode="showOnDemand" />
 					<widget source="Titel" render="Label" position="20,3" zPosition="3" size="680,50" font="Regular;20" valign="center" halign="left" transparent="1" foregroundColor="#ffffff"/>
 					<widget source="Titel2" render="Label" position="40,5" zPosition="2" size="640,50" font="Regular;22" valign="center" halign="center" transparent="1" foregroundColor="#f47d19"/>
 					<eLabel position="5,55" zPosition="2" size="710,2" foregroundColor="#c3c3c9" backgroundColor="#FFFFFF" />
@@ -674,6 +680,7 @@ class ForecaPreview(Screen, HelpableScreen):
 			self["key_blue"] = StaticText(_("Home"))
 		self["key_info"] = StaticText(_("Legend"))
 		self["key_menu"] = StaticText(_("Maps"))
+		self["key_help"] = StaticText(_("International City"))
 		self.setTitle(_("Foreca Weather Forecast") + "   " + _("Version ") + VERSION)
 
 		HelpableScreen.__init__(self)
@@ -682,7 +689,8 @@ class ForecaPreview(Screen, HelpableScreen):
 				"cancel": (self.exit, _("Exit - End")),
 				"menu": (self.Menu, _("Menu - Weather maps")),
 				"showEventInfo": (self.info, _("Info - Legend")),
-				"ok": (self.OK, _("OK - City")),
+				"ok": (self.Greek, _("OK - City")),
+				"contextMenu": (self.OK, _("Help - International")),
 				"left": (self.left, _("Left - Previous day")),
 				"right": (self.right, _("Right - Next day")),
 				"up": (self.up, _("Up - Previous page")),
@@ -874,7 +882,13 @@ class ForecaPreview(Screen, HelpableScreen):
 
 	def OK(self):
 		global city
-		panelmenu = ""
+		panelmenu = "/City.cfg"
+		city = self.ort
+		self.session.openWithCallback(self.OKCallback, CityPanel,panelmenu)
+
+	def Greek(self):
+		global city
+		panelmenu = "/GreekCity.cfg"
 		city = self.ort
 		self.session.openWithCallback(self.OKCallback, CityPanel, panelmenu)
 
@@ -1168,6 +1182,7 @@ class CityPanel(Screen, HelpableScreen):
 		self.setup_title = _("Select a city")
 		self.Mlist = []
 		self["Mlist"] = CityPanelList([])
+		self.panelmenu = panelmenu
 
 		self.onChangedEntry = []
 
@@ -1199,8 +1214,8 @@ class CityPanel(Screen, HelpableScreen):
 
 	def prepare(self):
 		self.maxidx = 0
-		if fileExists(USR_PATH + "/City.cfg"):
-			file = open(USR_PATH + "/City.cfg", "r", encoding="UTF-8")
+		if fileExists(USR_PATH + self.panelmenu):
+			file = open(USR_PATH + self.panelmenu, "r", encoding="UTF-8")
 			for line in file:
 				text = line.strip()
 				self.maxidx += 1
@@ -1553,7 +1568,10 @@ class SatPanel(Screen, HelpableScreen):
 #------------------------------------------------------------------------------------------
 
 	def SatEntryItem(self, entry):
+		flags = 0
 		pict_scale = self["Mlist"].pictScale
+		if pict_scale:
+			flags = BT_SCALE # need stretch to height, due it is missing BT_KEEP_ASPECT_RATIO
 
 		ItemSkin = self["Mlist"].itemHeight
 
@@ -1610,7 +1628,6 @@ class SatPanel(Screen, HelpableScreen):
 				f.write(resp.read())
 
 			self.session.open(PicView, devicepath, 0, False)
-
 		else:
 			# http://www.foreca.de/Austria/Linz?map=sat
 			devicepath = "/tmp/sat.html"
diff --git a/setup.py b/setup.py
index be0878a..0a39366 100644
--- a/setup.py
+++ b/setup.py
@@ -8,6 +8,6 @@ setup(name='enigma2-plugin-extensions-foreca',
        packages=[pkg],
        package_dir={pkg: 'plugin'},
        package_data={pkg: ['*.png', '*.xml', '*/*.png', 'locale/*/LC_MESSAGES/*.mo']},
-       data_files=[('/etc/enigma2/Foreca', ['plugin/City.cfg', 'plugin/fav1.cfg', 'plugin/fav2.cfg', 'plugin/startservice.cfg', 'plugin/Filter.cfg'])],
+       data_files=[('/etc/enigma2/Foreca', ['plugin/City.cfg', 'plugin/GreekCity.cfg', 'plugin/fav1.cfg', 'plugin/fav2.cfg', 'plugin/startservice.cfg', 'plugin/Filter.cfg'])],
        cmdclass=setup_translate.cmdclass, # for translation
       )
