diff --git a/IPTVPlayer/setup/iptvsetupimpl.py b/IPTVPlayer/setup/iptvsetupimpl.py
index 9ad17606..e9c036db 100644
--- a/IPTVPlayer/setup/iptvsetupimpl.py
+++ b/IPTVPlayer/setup/iptvsetupimpl.py
@@ -110,7 +110,7 @@ class IPTVSetupImpl:
 
         # cmdwrap
         self.cmdwrapVersion = 2
-        self.cmdwrapPaths = [resolveFilename(SCOPE_PLUGINS, 'Extensions/IPTVPlayer/bin/cmdwrap'), "/usr/bin/cmdwrap"]
+        self.cmdwrapPaths = [resolveFilename(SCOPE_PLUGINS, 'Extensions/IPTVPlayer/bin/cmdwrap'), "/usr/bin/cmdwrapper"]
 
         # duk
         self.dukVersion = 6 # "2.1.99 [experimental]" # real version
@@ -306,7 +306,7 @@ class IPTVSetupImpl:
     def getOpensslVersionFinished(self, stsTab, dataTab):
         printDBG("IPTVSetupImpl.getOpensslVersionFinished")
         if len(stsTab) > 0 and True == stsTab[-1]:
-            for ver in ['0.9.8', '1.0.0', '1.0.2']:
+            for ver in ['0.9.8', '1.0.0', '1.0.2', '1.1.1']:
                 if ver in dataTab[-1]:
                     self.openSSLVersion = '.' + ver
                     break
@@ -929,7 +929,7 @@ class IPTVSetupImpl:
         printDBG("IPTVSetupImpl.cmdwrapStep")
 
         def _detectValidator(code, data):
-            if 'cmdwrap input_file' in data:
+            if 'cmdwrapper input_file' in data:
                 try:
                     tmp = re.search("Version\:\s*?([0-9.]+?)[^0-9^.]", data).group(1)
                     if float(tmp) >= self.cmdwrapVersion:
diff --git a/IPTVPlayer/tools/e2ijs.py b/IPTVPlayer/tools/e2ijs.py
index 84d50239..669c2a7c 100644
--- a/IPTVPlayer/tools/e2ijs.py
+++ b/IPTVPlayer/tools/e2ijs.py
@@ -12,7 +12,7 @@ from hashlib import md5
 import time
 import thread
 
-DUKTAPE_VER = '226'
+DUKTAPE_VER = '250'
 
 
 def duktape_execute(cmd_params):
