From 7cf9b7f3a64b2d6d1bb541fe3902fdd4dfaa4102 Mon Sep 17 00:00:00 2001
From: Athanasios Oikonomou <athoik@gmail.com>
Date: Wed, 18 Sep 2013 20:52:58 +0300
Subject: [PATCH] Satdreamgr Changes


diff --git a/plugin/plugin.py b/plugin/plugin.py
index 8f5d084..af1edeb 100644
--- a/plugin/plugin.py
+++ b/plugin/plugin.py
@@ -12,8 +12,8 @@ url_sc = "/usr/lib/enigma2/python/Plugins/Extensions/GreekStreamTV/update.sh"
 url_pd = "/usr/lib/enigma2/python/Plugins/Extensions/GreekStreamTV/depends.sh"
 
 def menu(menuid, **kwargs):
-    if menuid == "mainmenu":
-        return [("GreekStreamTV", main, "GreekStreamTV", 33)]
+    if menuid == "none":
+        return [("GreekStreamTV", main, "GreekStreamTV", 45)]
     return []
 
 def main(session, **kwargs):
-- 
1.7.10.4

