From eb7098b8bf596e7a487a5f1918090e32b98f9ecd Mon Sep 17 00:00:00 2001
From: Athanasios Oikonomou <athoik@gmail.com>
Date: Wed, 18 Sep 2013 20:52:58 +0300
Subject: [PATCH] Satdreamgr Changes


diff --git a/plugin/plugin.py b/plugin/plugin.py
index ee1744a..db28ec2 100644
--- a/plugin/plugin.py
+++ b/plugin/plugin.py
@@ -27,8 +27,8 @@ GSBQ = "/etc/enigma2/userbouquet.greekstreamtv.tv"
 
 
 def menu(menuid, **kwargs):
-    if menuid == "mainmenu":
-        return [("GreekStreamTV", main, "GreekStreamTV", 33)]
+    if menuid == "none":
+        return [("GreekStreamTV", main, "GreekStreamTV", 45)]
     return []
 
 
-- 
2.1.4

