Subject: [PATCH] Set default skin Satdreamgr-HD-TranspBA

---
 skin.py |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: git/skin.py
===================================================================
--- git.orig/skin.py
+++ git/skin.py
@@ -76,7 +76,7 @@ def skin_user_skinname():
 
 # example: loadSkin("nemesis_greenline/skin.xml")
 config.skin = ConfigSubsection()
-DEFAULT_SKIN = SystemInfo["HasFullHDSkinSupport"] and "PLi-FullNightHD/skin.xml" or "PLi-HD/skin.xml"
+DEFAULT_SKIN = "Satdreamgr-HD-TranspBA/skin.xml"
 # on SD hardware, PLi-HD will not be available
 if not fileExists(resolveFilename(SCOPE_SKIN, DEFAULT_SKIN)):
 	# in that case, fallback to Magic (which is an SD skin)
