diff --git a/lib/python/Screens/ChannelSelection.py b/lib/python/Screens/ChannelSelection.py
index 2b8d9da735..b4ffb6a579 100644
--- a/lib/python/Screens/ChannelSelection.py
+++ b/lib/python/Screens/ChannelSelection.py
@@ -285,6 +285,10 @@ class ChannelContextMenu(Screen):
 								append_when_current_valid(current, menu, (_("Do not center DVB subs on this service"), self.removeCenterDVBSubsFlag), level=2)
 							else:
 								append_when_current_valid(current, menu, (_("Do center DVB subs on this service"), self.addCenterDVBSubsFlag), level=2)
+
+					if not eDVBDB.getInstance().isCrypted(eServiceReference(current.toString())):
+						append_when_current_valid(current, menu, (_("Add Biss Caid to service"), self.setBISS), level=0)
+
 					if haveBouquets:
 						bouquets = self.csel.getBouquetList()
 						if bouquets is None:
@@ -400,6 +404,14 @@ class ChannelContextMenu(Screen):
 		else:
 			self.insertService()
 
+	def setCAID(self, value):
+		eDVBDB.getInstance().addCAID(eServiceReference(self.csel.getCurrentSelection().toString()), value)
+		eDVBDB.getInstance().reloadBouquets()
+		self.close()
+
+	def setBISS(self):
+		self.setCAID(0x2600)
+
 	def set3DMode(self, value):
 		playingref = self.session.nav.getCurrentlyPlayingServiceReference()
 		if config.plugins.OSD3DSetup.mode.value == "auto" and (playingref and playingref == self.csel.getCurrentSelection()):
