diff --git a/lib/python/Components/ServicePosition.py b/lib/python/Components/ServicePosition.py
index 04121a340e..1c1b845fc8 100644
--- a/lib/python/Components/ServicePosition.py
+++ b/lib/python/Components/ServicePosition.py
@@ -63,7 +63,7 @@ class ServicePosition(PerServiceDisplay, object):
 				elif self.type == self.TYPE_REMAINING:
 					l = self.get(self.TYPE_LENGTH) - self.get(self.TYPE_POSITION)
 
-				self.setText("%d:%02d" % (l / 60, l % 60))
+				self.setText("%2d:%02d" % (l / 60, l % 60))
 			else:
 				l = self.get(self.TYPE_POSITION)
 				if l != -1:
