diff --git a/lib/python/Components/SystemInfo.py b/lib/python/Components/SystemInfo.py
index dc5521d530..ae72d29e57 100644
--- a/lib/python/Components/SystemInfo.py
+++ b/lib/python/Components/SystemInfo.py
@@ -140,6 +140,7 @@ from Tools.Multiboot import getMultibootStartupDevice, getMultibootslots  # This
 
 def setBoxInfoItems():
 	model = BoxInfo.getItem("machine")
+	BoxInfo.setItem("HDMIOutput", fileCheck("/proc/stb/hdmi/output"))
 	BoxInfo.setItem("InDebugMode", eGetEnigmaDebugLvl() >= 4)
 	BoxInfo.setItem("CommonInterface", model in ("h9combo", "h9combose", "h10", "pulse4kmini") and 1 or eDVBCIInterfaces.getInstance().getNumOfSlots())
 	BoxInfo.setItem("CommonInterfaceCIDelay", fileCheck("/proc/stb/tsmux/rmx_delay"))
diff --git a/lib/python/Screens/Standby.py b/lib/python/Screens/Standby.py
index 214706c922..e5a8c9b60a 100644
--- a/lib/python/Screens/Standby.py
+++ b/lib/python/Screens/Standby.py
@@ -58,6 +58,12 @@ class StandbyScreen(Screen):
 		if os.path.exists("/usr/script/standby_enter.sh"):
 			Console().ePopen("/usr/script/standby_enter.sh")
 
+		if BoxInfo.getItem("HDMIOutput"):
+			try:
+				open(BoxInfo.getItem("HDMIOutput"), "w").write("off")
+			except:
+				pass
+
 		self["actions"] = ActionMap(["StandbyActions"],
 		{
 			"power": self.Power,
@@ -152,6 +158,13 @@ class StandbyScreen(Screen):
 		self.leaveMute()
 		if os.path.exists("/usr/script/standby_leave.sh"):
 			Console().ePopen("/usr/script/standby_leave.sh")
+
+		if BoxInfo.getItem("HDMIOutput"):
+			try:
+				open(BoxInfo.getItem("HDMIOutput"), "w").write("on")
+			except:
+				pass
+
 		if config.usage.remote_fallback_import_standby.value:
 			ImportChannels()
 
