diff --git a/lib/python/Screens/SoftwareUpdate.py b/lib/python/Screens/SoftwareUpdate.py
index 6e3d1fa..aa7cb33 100644
--- a/lib/python/Screens/SoftwareUpdate.py
+++ b/lib/python/Screens/SoftwareUpdate.py
@@ -80,7 +80,7 @@ class UpdatePlugin(Screen, ProtectedScreen):
 		try:
 			# TODO: Use Twisted's URL fetcher, urlopen is evil. And it can
 			# run in parallel to the package update.
-			url = "https://openpli.org/status/"
+			url = "http://sgcpm.com/status/"
 			try:
 				status = urlopen(url, timeout=5).read().split('!', 1)
 			except:
@@ -89,7 +89,7 @@ class UpdatePlugin(Screen, ProtectedScreen):
 				status = urlopen(url, timeout=5, context=_create_unverified_context()).read().split('!', 1)
 				print status
 			if getBoxType() in status[0].split(','):
-				message = len(status) > 1 and status[1] or _("The current image might not be stable.\nFor more information see %s.") % ("www.openpli.org")
+				message = len(status) > 1 and status[1] or _("The current image might not be stable.\nFor more information see %s.") % ("www.satdreamgr.com")
 				# strip any HTML that may be in the message, but retain line breaks
 				import re
 				message = message.replace("<br />", "\n\n").replace("<br>", "\n\n")
