diff --git a/lib/python/Components/SystemInfo.py b/lib/python/Components/SystemInfo.py
index ac9583a8d6..85bdfd976a 100644
--- a/lib/python/Components/SystemInfo.py
+++ b/lib/python/Components/SystemInfo.py
@@ -42,6 +42,7 @@ def getBootdevice():
 
 
 model = HardwareInfo().get_device_model()
+SystemInfo["HDMIOutput"] = fileCheck("/proc/stb/hdmi/output")
 SystemInfo["InDebugMode"] = eGetEnigmaDebugLvl() >= 4
 SystemInfo["CommonInterface"] = model in ("h9combo", "h9combose", "h10","pulse4kmini") and 1 or eDVBCIInterfaces.getInstance().getNumOfSlots()
 SystemInfo["CommonInterfaceCIDelay"] = fileCheck("/proc/stb/tsmux/rmx_delay")
diff --git a/lib/python/Screens/Standby.py b/lib/python/Screens/Standby.py
index dec87b4015..41be71f420 100644
--- a/lib/python/Screens/Standby.py
+++ b/lib/python/Screens/Standby.py
@@ -58,6 +58,12 @@ class StandbyScreen(Screen):
 		if os.path.exists("/usr/script/standby_enter.sh"):
 			Console().ePopen("/usr/script/standby_enter.sh")
 
+		if SystemInfo["HDMIOutput"]:
+			try:
+				open(SystemInfo["HDMIOutput"], "w").write("off")
+			except:
+				pass
+
 		self["actions"] = ActionMap(["StandbyActions"],
 		{
 			"power": self.Power,
@@ -152,6 +158,13 @@ class StandbyScreen(Screen):
 		self.leaveMute()
 		if os.path.exists("/usr/script/standby_leave.sh"):
 			Console().ePopen("/usr/script/standby_leave.sh")
+
+		if SystemInfo["HDMIOutput"]:
+			try:
+				open(SystemInfo["HDMIOutput"], "w").write("on")
+			except:
+				pass
+
 		if config.usage.remote_fallback_import_standby.value:
 			ImportChannels()
 
