diff --git a/lib/python/Components/HdmiCec.py b/lib/python/Components/HdmiCec.py
index 8646abbf7c..e540a76e25 100644
--- a/lib/python/Components/HdmiCec.py
+++ b/lib/python/Components/HdmiCec.py
@@ -104,6 +104,7 @@ class HdmiCec:
 		assert not HdmiCec.instance, "only one HdmiCec instance is allowed!"
 		HdmiCec.instance = self
 
+		self.lastCmd = -1
 		self.wait = eTimer()
 		self.wait.timeout.get().append(self.sendCmd)
 		self.waitKeyEvent = eTimer()
@@ -329,6 +330,13 @@ class HdmiCec:
 
 			if config.hdmicec.debug.value != "0":
 				self.debugRx(length, cmd, data)
+
+			if cmd == self.lastCmd:
+				print("eHdmiCec: ignore duplicate cmd %02X" % cmd)
+				return
+			else:
+				self.lastCmd = cmd
+
 			if cmd == 0x00:
 				if length == 0: # only polling message ( it's some as ping )
 					print("eHdmiCec: received polling message")
