diff --git a/lib/python/Screens/FlashImage.py b/lib/python/Screens/FlashImage.py
index 8c88bc9038..7f040db216 100644
--- a/lib/python/Screens/FlashImage.py
+++ b/lib/python/Screens/FlashImage.py
@@ -96,9 +96,10 @@ class SelectImage(Screen):
 
 		if not self.imagesList:
 			if not self.jsonlist:
-				url = "http://downloads.openpli.org/json/%s" % model
+				url = "http://sgcpm.com/json/%s" % model
 				try:
-					self.jsonlist = dict(json.load(urlopen(url, timeout=15)))
+					req = Request(url, None, {"User-agent": "Enigma2 SDG"})
+					self.jsonlist = dict(json.load(urlopen(req, timeout=15)))
 				except:
 					print("[FlashImage] getImagesList Error: Unable to load json data from URL '%s'!" % url)
 				alternative_imagefeed = config.usage.alternative_imagefeed.value
