diff --git a/lib/python/Screens/FlashImage.py b/lib/python/Screens/FlashImage.py
index a48669e9e6..49ed7bb616 100644
--- a/lib/python/Screens/FlashImage.py
+++ b/lib/python/Screens/FlashImage.py
@@ -83,7 +83,7 @@ class SelectImage(Screen):
 		if not self.imagesList:
 			if not self.jsonlist:
 				try:
-					self.jsonlist = dict(json.load(urlopen('http://downloads.openpli.org/json/%s' % model)))
+					self.jsonlist = dict(json.load(urlopen('http://sgcpm.com/json/%s' % model, headers={'User-Agent': 'Enigma2 SDG'})))
 					if config.usage.alternative_imagefeed.value:
 						self.jsonlist.update(dict(json.load(urlopen('%s%s' % (config.usage.alternative_imagefeed.value, model)))))
 				except:
