diff --git a/lib/python/Screens/FlashImage.py b/lib/python/Screens/FlashImage.py
index 42c97be891..59309e9365 100644
--- a/lib/python/Screens/FlashImage.py
+++ b/lib/python/Screens/FlashImage.py
@@ -81,7 +81,7 @@ class SelectImage(Screen):
 		if not self.imagesList:
 			if not self.jsonlist:
 				try:
-					self.jsonlist = dict(json.load(urllib.request.urlopen('http://downloads.openpli.org/json/%s' % model)))
+					self.jsonlist = dict(json.load(urllib.request.urlopen('http://sgcpm.com/json/%s' % model, headers={'User-Agent': 'Enigma2 SDG'})))
 					if config.usage.alternative_imagefeed.value:
 						self.jsonlist.update(dict(json.load(urllib.request.urlopen('%s%s' % (config.usage.alternative_imagefeed.value, model)))))
 				except:
