diff --git a/lib/python/Screens/FlashImage.py b/lib/python/Screens/FlashImage.py
index 320bfd7ecf..a004b0ad71 100644
--- a/lib/python/Screens/FlashImage.py
+++ b/lib/python/Screens/FlashImage.py
@@ -43,7 +43,7 @@ class SelectImage(Screen):
 		self.setIndex = 0
 		self.expanded = []
 		self.model = HardwareInfo().get_machine_name()
-		self.selectedImage = ["OpenPLi", {"url": "https://downloads.openpli.org/json/%s" % self.model, "model": self.model}]
+		self.selectedImage = ["SDG", {"url": "http://sgcpm.com/json/%s" % self.model, "model": self.model}]
 		self.models = [self.model]
 		self.setTitle(_("Select image"))
 		self["key_red"] = StaticText(_("Cancel"))
@@ -113,7 +113,8 @@ class SelectImage(Screen):
 		if not self.imagesList:
 			if not self.jsonlist:
 				try:
-					self.jsonlist = dict(json.load(urlopen(self.selectedImage[1]["url"], timeout=3)))
+					req = Request(self.selectedImage[1]["url"], None, {"User-agent": "Enigma2 SDG"})
+					self.jsonlist = dict(json.load(urlopen(req, timeout=15)))
 				except:
 					print("[FlashImage] getImagesList Error: Unable to load json data from URL '%s'!" % self.selectedImage[1]["url"])
 				alternative_imagefeed = config.usage.alternative_imagefeed.value
