diff --git a/lib/python/Screens/InputDeviceSetup.py b/lib/python/Screens/InputDeviceSetup.py
index 68c1391859..e89de95553 100644
--- a/lib/python/Screens/InputDeviceSetup.py
+++ b/lib/python/Screens/InputDeviceSetup.py
@@ -293,6 +293,7 @@ class InputDeviceSetup(Screen, ConfigListScreen):
 class RemoteControlType(Screen, ConfigListScreen):
 	rcList = [
 			("0", _("Default")),
+			("501", _("Octagon SF4008")),
 			("4", _("DMM normal")),
 			("5", _("et9000/et9100")),
 			("6", _("DMM advanced")),
@@ -314,6 +315,7 @@ class RemoteControlType(Screen, ConfigListScreen):
 		]
 
 	defaultRcList = [
+			("sf4008", 501),
 			("et4000", 13),
 			("et5000", 7),
 			("et6000", 7),
