diff --git a/lib/python/Screens/InputDeviceSetup.py b/lib/python/Screens/InputDeviceSetup.py
index b012300..702de51 100644
--- a/lib/python/Screens/InputDeviceSetup.py
+++ b/lib/python/Screens/InputDeviceSetup.py
@@ -312,7 +312,8 @@ class RemoteControlType(Screen, ConfigListScreen):
 			("19", _("HD2400")),
 			("20", _("Zgemma Star S/2S/H1/H2")),
 			("21", _("Zgemma H.S/H.2S/H.2H/H5/H7")),
-			("25", _("Zgemma H9/i55 Plus"))
+			("25", _("Zgemma H9/i55 Plus")),
+			("501", _("Octagon SF4008")),
 		]
 
 	defaultRcList = [
@@ -349,7 +350,8 @@ class RemoteControlType(Screen, ConfigListScreen):
 			("h5", 21),
 			("h7", 21),
 			("h9", 25),
-			("et7000mini", 16)
+			("et7000mini", 16),
+			("sf4008", 501),
 		]
 
 	def __init__(self, session):
