diff --git a/lib/dvb/frontend.cpp b/lib/dvb/frontend.cpp
index 8a8677e..52d41df 100644
--- a/lib/dvb/frontend.cpp
+++ b/lib/dvb/frontend.cpp
@@ -1352,6 +1352,7 @@ int eDVBFrontend::readFrontendData(int type)
 									strength = prop[0].u.st.stat[i].uvalue;
 							}
 						}
+						if (!strength && m_type == iDVBFrontend::feTerrestrial) return strength;
 					}
 #endif
 					// fallback to old DVB API
